Sources/Core/Needs Conversion/Logger.swift

```diff
@@ -18,12 +18,22 @@
 
 import Foundation
 
-// This will ultimately be a wrapper around os_log. For more info on the decision to
-// rewrite this to wrap os_log over NSLog see: https://developer.apple.com/videos/play/wwdc2016/721/
-// more info here: https://stackoverflow.com/questions/25951195/swift-print-vs-println-vs-nslog
-//
 struct Logger: Logging {
-  func log(message: StaticString) {
-    // TODO: Implementation
+  let settings: SettingsManaging
+
+  init(settings: SettingsManaging = Settings.shared) {
+    self.settings = settings
+  }
+
+  func log(message: String, for behavior: LoggingBehavior) {
+    guard isAbleToLog(behavior) else {
+      return
+    }
+
+    NSLog(message)
```

Comment:
So... NSLog still have some benefits that print does not. Primarily that it logs to a file so that it's readable by the console app (the most useful tool that nobody uses). I'm pretty sure there's no way to figure out how many users of the sdk use the console app for debugging so didn't want to break it for any that do.